File: grok.js
const start = Date.now()* 1_000;


const baseline = Array(30).fill("8");

const toChange = [
  [1],
  [],
  [13, 14, 15, 16, 24, 25],
  [9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 22, 23, 24],
  [7, 8, 9, 10, 11, 12, 20, 21, 22, 23],
  [6, 7, 8, 9, 18, 19, 20, 21, 22, 23, 24],
  [6, 7, 8, 16, 17, 18, 21, 22, 23],
  [6, 7, 8, 14, 15, 16, 22, 23, 24],
  [6, 7, 8, 9, 21, 22, 23],
  [7, 8, 9, 18, 19, 20, 21, 22, 23],
  [6, 7, 8, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21],
  [4, 5, 11, 12, 13, 14, 15, 16, 17, 18],
  [],
  [-1]
];

for (let t = 0; t < 1000000; t++) {
for (let i = 0; i < 14; i++) {
  const thisLine = [...baseline]; 
  for (const index of toChange[i]) {
    if (index >= 0) {
      thisLine[index] = i === 0 ? "0" : "1";
    }
  }
  
}}

const end = Date.now()* 1_000;
const timeTakenMicro = end - start;
const timeTakenMilli = timeTakenMicro/ 1_000;
const timeTakenSecs = timeTakenMicro/ 1_000_000;

console.log("Time taken in microseconds:", timeTakenMicro.toString());
console.log("Time taken in milliseconds:", timeTakenMilli.toString());
console.log("Time taken in seconds:", timeTakenSecs);

File: grok.py
from datetime import datetime, timezone

start = int(datetime.now(timezone.utc).timestamp() * 1_000_000)
# arr of 8 in 30
baseline = ["8"] * 30
to_change = [
    [1],
    [],
    list(range(13,17)) + list(range(24,26)),
    list(range(9,20)) + list(range(22, 25)),
    list(range(7, 13))+ list(range(20,24)),
    list(range(6,10)) + list(range(18, 25)),
    list(range(6,9)) + list(range(16, 19)) + list(range(21,24)),
    list(range(6,9)) + list(range(14,17)) + list(range(22, 25)),
    list(range(6,10)) + list(range(21, 24)),
    list(range(7,10)) + list(range(18,24)),
    list(range(6,10)) + list(range(10, 22)),
    list(range(4,6)) + list(range(11, 19)),
    [],[-1]
]
for t in range(0,1000000):
    for i in range(0,14):
        this_line = baseline.copy()
        for each in to_change[i]:
            this_line[each] = 1 if i != 0 else 0
    
end = int(datetime.now(timezone.utc).timestamp() * 1_000_000)
time_taken = end -  start
print("Time taken in micro seconds", time_taken)
print("Time taken in milli seconds", time_taken/1000)
print("Time taken in seconds", time_taken/1_000_000)